src/Workspaces/Core/Portable/Execution/AbstractReferenceSerializationService.cs

```diff
@@ -92,6 +94,14 @@ public void WriteTo(AnalyzerReference reference, ObjectWriter writer, Cancellati
             var file = reference as AnalyzerFileReference;
             if (file != null)
             {
+                // fail to load analyzer assembly
+                var assemblyPath = TryGetAnalyzerAssemblyPath(file);
```

Comment:
that is display path. assembly path is path of actual assembly we are going to load (such as shadow copied assembly)

assembly path comes from reference.GetAssembly().Location which will throw if it couldn't actually load assembly because it doesnt exist.